indico/web/client/js/react/components/EditableSubmissionButton.jsx

```diff
@@ -0,0 +1,71 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {useCallback, useState} from 'react';
+import PropTypes from 'prop-types';
+import {Button, Modal} from 'semantic-ui-react';
+import FileManager from 'indico/modules/events/editing/components/FileManager';
+import {fileTypePropTypes} from 'indico/modules/events/editing/components/FileManager/util';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+
+export default function EditableSubmissionButton({
+  eventId,
+  contributionId,
+  uploadURL,
+  fileTypes,
+  type,
+  submitRevisionURL,
+}) {
+  const [submissionFiles, setSubmissionFiles] = useState({});
+  const [submitting, setSubmitting] = useState(false);
+  async function submitRevision() {
+    setSubmitting(true);
+    try {
+      await indicoAxios.put(submitRevisionURL, {
+        files: submissionFiles,
+      });
+    } catch (e) {
+      handleAxiosError(e, false, true);
+      setSubmitting(false);
+      return;
+    }
+    setSubmitting(false);
+    window.location.href = `/event/${eventId}/contributions/${contributionId}/editing/${type}`;
+  }
+
+  const handleChange = useCallback(files => setSubmissionFiles(files), []);
+  return (
+    <Modal
+      trigger={
+        <button type="submit" className="i-button highlight">
+          Submit Files
+        </button>
+      }
+      closeIcon
+    >
+      <Modal.Header>Submit a {type}</Modal.Header>
```

Comment:
i18n and you have to explicitly write two strings for paper and slides since interpolating the `{type}` in the translated string won't work for anything but english.

"Submit your paper" and "Submit your slides" are probably good texts for it